tools/analyzer_plugin/lib/src/util/prop_declarations/get_all_props.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:over_react_analyzer_plugin/src/util/util.dart';
+
+// Performance optimization notes:
```

Comment:
I spent quite a bit of time optimizing this function, and I feel pretty happy about its performance! 

In practice, usage is optimized further via the caching done with `_cachedGetAllRequiredProps`.